synapse/streams/config.py

```diff
@@ -78,29 +66,22 @@ def from_request(cls, request, raise_invalid_params=True, default_limit=None):
 
         limit = parse_integer(request, "limit", default=default_limit)
 
-        if limit and limit < 0:
-            raise SynapseError(400, "Limit must be 0 or above")
+        if limit:
+            if limit < 0:
+                raise SynapseError(400, "Limit must be 0 or above")
+
+            limit = min(int(limit), MAX_LIMIT)
```

Comment:
Limiting this looks like it was previously done in `SourcePaginationConfig`, was there anything that used `PaginationConfig` that wouldn't want this set? (Is this a change in a behavior?)

Reply:
Doh, I see this was done in the `__init__`. Ignore this comment.